src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -47,12 +47,31 @@ type dist = [
   | `Cauchy(cauchy)
   | `Triangular(triangular)
   | `ContinuousShape(continuousShape)
-  | `Float(float)
+  | `Float(float) // Dirac delta at x. Practically useful only in the context of multimodals.
 ];
 
-type pointwiseAdd = array((dist, float));
+type integral = float;
+type cutoffX = float;
+type operation = [
+    | `AddOperation
+    | `SubtractOperation
+    | `MultiplyOperation
+    | `DivideOperation
+    | `ExponentiateOperation
+];
 
-type bigDist = [ | `Simple(dist) | `PointwiseCombination(pointwiseAdd)];
+type distTree = [
+    | `Simple(dist)
+    | `Combination(distTree, distTree, operation)
+    | `PointwiseSum(distTree, distTree)
+    | `PointwiseProduct(distTree, distTree)
+    | `VerticalScaling(distTree, distTree)
+    | `Normalize(distTree)
+    | `LeftTruncate(distTree, cutoffX)
+    | `RightTruncate(distTree, cutoffX)
+    | `Render(distTree)
+]
+and weightedDists = array((distTree, float));
```

Comment:
does this have to be a separate type from distTree? I don't see it being used elsewhere.